OpenBSD 6.3-CURRENT (amd64) fails building on sqlite.

```
../../../src/std/db/_sqlite.c:626:10: fatal error: 'sqlite3.h' file not found
#include <sqlite3.h>
         ^~~~~~~~~~~
1 error generated.
*** ERROR IN ##main -- target compilation or link failed while compiling "/home/jamppa/git/gerbil/src/std/db/_sqlite.scm"
*** ERROR IN std/make#gsc-compile, "make.ss"@416.7 -- Compilation error; gsc exited with nonzero status 17920
*** ERROR; build failed
```

Currently running on `OpenBSD kaganovich.my.domain 6.3 GENERIC.MP#25 amd64` and both `sqlite`'s are found on the system.

```
sqlite-2.8.17p4 (installed)
sqlite3-3.23.0 (installed)
```

`sqlite3.h` is located in `/usr/local/include/sqlite3.h`. I think it could be as easy as allowing a compiler/linker-flag to include that directory in its' search path, but can't seem to figure out how.

If I personally write out the whole db-part out of files/features to build, `gerbil` builds without a hitch.

--- a/src/std/build-spec.ss
+++ b/src/std/build-spec.ss
@@ -1,8 +1,4 @@
 ;; -*- Gerbil -*-
-
-(def ldflags (env-ldflags))
-(def cppflags (env-cppflags))
-
 (def build-spec
   `((gxc: "build-config" dep: ("build-features.ss"))
     "format"
@@ -62,15 +58,15 @@
     "text/json"
     ,@(if config-enable-libyaml
         `((gsc: "text/libyaml"
-                "-cc-options" ,(cppflags "")
-                "-ld-options" ,(ldflags "-lyaml"))
+                "-cc-options" ,(cppflags "libyaml" "")
+                "-ld-options" ,(ldflags "libyaml" "-lyaml"))
           (ssi: "text/libyaml")
           "text/yaml")
         '())
     ,@(if config-enable-zlib
         `((gsc: "text/_zlib"
-                "-cc-options" ,(cppflags "")
-                "-ld-options" ,(ldflags "-lz"))
+                "-cc-options" ,(cppflags "zlib" "")
+                "-ld-options" ,(ldflags "zlib" "-lz"))
           (ssi: "text/_zlib")
           "text/zlib")
         '())
@@ -146,8 +142,8 @@
     "xml"
     ;; :std/crypto
     (gsc: "crypto/libcrypto"
-          "-cc-options" ,(cppflags "")
-          "-ld-options" ,(ldflags "-lcrypto")
+          "-cc-options" ,(cppflags "libcrypto" "")
+          "-ld-options" ,(ldflags "libcrypto" "-lcrypto")
           ,@(include-gambit-sharp))
     (static-include: "crypto/libcrypto-rfc5114.c")
     (ssi: "crypto/libcrypto")
@@ -197,29 +193,29 @@
     "db/postgresql"
     ,@(if config-enable-sqlite
         `((gsc: "db/_sqlite"
-                "-cc-options" ,(cppflags "")
-                "-ld-options" ,(ldflags "-lsqlite3"))
+                "-cc-options" ,(cppflags "sqlite3" "")
+                "-ld-options" ,(ldflags "sqlite3" "-lsqlite3"))
           (ssi: "db/_sqlite")
           "db/sqlite")
         '())
     ,@(if config-enable-mysql
         `((gsc: "db/_mysql"
-                "-cc-options" ,(cppflags "")
-                "-ld-options" ,(ldflags "-lpthread -lmysqlclient"))
+                "-cc-options" ,(cppflags "mysqlclient" "")
+                "-ld-options" ,(ldflags "mysqlclient" "-lpthread -lmysqlclient"))
           (ssi: "db/_mysql")
           "db/mysql")
         '())
     ,@(if config-enable-lmdb
         `((gsc: "db/_lmdb"
-                "-cc-options" ,(cppflags "")
-                "-ld-options" ,(ldflags "-llmdb"))
+                "-cc-options" ,(cppflags "lmdb" "")
+                "-ld-options" ,(ldflags "lmdb" "-llmdb"))
           (ssi: "db/_lmdb")
           "db/lmdb")
         '())
     ,@(if config-enable-leveldb
         `((gsc: "db/_leveldb"
-                "-cc-options" ,(cppflags "")
-                "-ld-options" ,(ldflags "-lleveldb"))
+                "-cc-options" ,(cppflags "leveldb" "")
+                "-ld-options" ,(ldflags "leveldb" "-lleveldb"))
           (ssi: "db/_leveldb")
           "db/leveldb")
         '())

--- a/src/std/make.ss
+++ b/src/std/make.ss
@@ -612,15 +612,15 @@ package: std
   (string-join (map (lambda (l) (pkg-config l "--cflags")) libs) " "))
 
 ;; tries pkg-config-libs with fallback to env-ldflags
-(def (ldflags lib ldflags)
+(def (ldflags lib flags)
   (try
    (pkg-config-libs lib)
    (catch (e)
-     ((env-ldflags) ldflags))))
+     ((env-ldflags) flags))))
 
 ;; tries pkg-confg-cflags with fallback to env-cppflags
-(def (cppflags lib cppflags)
+(def (cppflags lib flags)
   (try
    (pkg-config-cflags lib)
    (catch (e)
-     ((env-cppflags) cppflags))))
+     ((env-cppflags) flags))))

--- a/src/std/make.ss
+++ b/src/std/make.ss
@@ -7,7 +7,8 @@ package: std
         :gerbil/expander
         :gerbil/gambit/misc
         :gerbil/gambit/ports
-        "sort")
+        "sort"
+        "sugar")
 (export make make-depgraph make-depgraph/spec
         shell-config
         env-cppflags
@@ -15,7 +16,9 @@ package: std
         include-gambit-sharp
         pkg-config
         pkg-config-libs
-        pkg-config-cflags)
+        pkg-config-cflags
+        ldflags
+        cppflags)
 
 ;; buildspec: [<build> ...]
 ;;  <build>:
@@ -592,7 +595,7 @@ package: std
 ;; args: [<arg> ...]
 ;;  <arg>: additionnal arguments to pass to pkg-config
 (def (pkg-config lib . args)
-  (apply shell-config "pkg-config" lib args))
+  (apply shell-config "pkg-config" "--silence-errors" lib args))
 
 ;; calls pkg-config in order to return `ld-options`.
 ;;
@@ -607,3 +610,17 @@ package: std
 ;;  <lib>: library to get cc-options for
 (def (pkg-config-cflags . libs)
   (string-join (map (lambda (l) (pkg-config l "--cflags")) libs) " "))
+
+;; tries pkg-config-libs with fallback to env-ldflags
+(def (ldflags lib ldflags)
+  (try
+   (pkg-config-libs lib)
+   (catch (e)
+     ((env-ldflags) ldflags))))
+
+;; tries pkg-confg-cflags with fallback to env-cppflags
+(def (cppflags lib cppflags)
+  (try
+   (pkg-config-cflags lib)
+   (catch (e)
+     ((env-cppflags) cppflags))))

